### PR TITLE
chore(sample-app): upgrade dependencies in sample app

### PR DESCRIPTION
**Pull request by vanGalilea**
## 🎯 Goal

Keeping our SA up to date

## 🛠 Implementation details

Upgrading dependencies in our SA

## 🎨 UI Changes

N/A

## 🧪 Testing

Run the SA on both OSs and validate chat features.

## ☑️ Checklist

- [ ] I have signed the [Stream CLA](https://docs.google.com/forms/d/e/1FAIpQLScFKsKkAJI7mhCr7K9rEIOpqIDThrWxuvxnwUq2XkHyG154vQ/viewform) (required)
- [ ] PR targets the `develop` branch
- [ ] Documentation is updated
- [ ] New code is tested in main example apps, including all possible scenarios
  - [ ] SampleApp iOS and Android
  - [ ] Expo iOS and Android

### CODE REVIEW

File: docusaurus/docs/reactnative/basics/troubleshooting.mdx [hunk 1]
```diff
@@ -350,6 +350,12 @@ Reanimated plugin has to be listed last.
 :::
 
 ### Blank screen on Android
+:::caution
+
+This step is only required for older version of (Reanimated 2).
+If you are using version >=3.1.0, you can skip this step.
```
Comment (santhoshvai) [inline docusaurus/docs/reactnative/basics/troubleshooting.mdx:356]: why not >= 3.0.. or I think its simpler for the reader if this section states that it is only relevant to reanimated v2 and not any other version

File: examples/SampleApp/Gemfile [hunk 2]
```diff
@@ -1,9 +1,9 @@
 source 'https://rubygems.org'
 
 # You may use http://rbenv.org/ or https://rvm.io/ to install and use this version
-ruby '2.7.5'
+ruby '>= 2.6.10'
```
Comment (santhoshvai) [inline examples/SampleApp/Gemfile:4]: why the downgrade? 

2.7.5 comes from here I think

https://github.com/react-native-community/react-native-template-typescript/blob/main/template/_ruby-version

File: examples/SampleApp/.node-version [hunk 3]
```diff
@@ -0,0 +1 @@
+18
```
Comment (santhoshvai) [inline examples/SampleApp/.node-version:1]: why 18? 

docusaurus will break I think

Comment (santhoshvai) [reply examples/SampleApp/.node-version:1]: but no problem in CI though.. locally will be a problem

Comment (santhoshvai) [reply examples/SampleApp/.node-version:1]: Ideally we update the docusaurus repo to support 18

Comment (vanGalilea) [reply examples/SampleApp/.node-version:1]: > Ideally we update the docusaurus repo to support 18

That's right, in the meanwhile, we should all use nvm to manage those versions.

**Pull request merged**